nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/nf-error-handler.js

```diff
@@ -142,6 +142,36 @@
                 // show the error pane
                 $('#message-pane').show();
             }
+        },
+
+        /**
+         * Method for handling ajax errors when submitting configuration update (PUT/POST) requests.
+         * In addition to what handleAjaxError does, this function splits
+         * the error message text to display them as an unordered list.
+         *
+         * @argument {object} xhr       The XmlHttpRequest
+         * @argument {string} status    The status of the request
+         * @argument {string} error     The error
+         */
+        handleConfigurationUpdateAjaxError: function (xhr, status, error) {
+            if (xhr.status === 400) {
+                var errors = xhr.responseText.split('\n');
+
+                var content;
+                if (errors.length === 1) {
+                    content = $('<span></span>').text(errors[0]);
+                } else {
+                    content = nfCommon.formatUnorderedList(errors);
+                }
+
+                nfDialog.showOkDialog({
+                    dialogContent: content,
+                    headerText: 'Configuration Error'
+                });
+            } else {
+                self.handleAjaxError(xhr, status, error);
```

Comment:
If this condition happens on the canvas (through `nf-canvas-error-handler`) I think we would actually want to invoke that `handleAjaxError` so that we can run those specific actions when we hide the canvas. This may not be possible from here so we may need additional logic in `nf-canvas-error-handler`.

Reply:
@mcgilman That's probably true. But I didn't see any logic that closes the canvas in the Javascript sources that I researched while I'm working on this PR, if I didn't oversight (that I do often though ... ><)

I named the function as `handleConfigurationUpdateAjaxError` to express that the function is expected to be used to handle errors while configuration updates.
Is there any situation that NiFi UI closes the canvas when configuration update fails?
Or would you share some specific code which closes the canvas?

Reply:
In `nf-error-handler.handleAjaxError` anywhere we show the `message-pane` the canvas will be hidden. Probably the most common scenario would be if your terminated your NiFi instance. The next request from the browser will fail and we show that message that indicates

> Please ensure the application is running and check the logs for any errors.

Reply:
The idea is the error handler is to simply show a dialog when the error is something that is expected and where we want the user to be able to continue using the application. When the error is not expected (like a 500 response code) or we do not want the user to attempt another request (like the user is not authenticated) we hide the canvas by showing the `message-pane`.

Reply:
Addressed this by extracting the actions to disable the canvas from `handleAjaxError` to a new function called `disableCanvas` in `nf-canvas-error-handler.js`, so that both `handleAjaxError` and `handleConfigurationUpdateAjaxError` can use it.